lib/activerecord-import/import.rb

```diff
@@ -846,5 +855,42 @@ def add_special_rails_stamps( column_names, array_of_attributes, options )
     def validations_array_for_column_names_and_attributes( column_names, array_of_attributes ) # :nodoc:
       array_of_attributes.map { |values| Hash[column_names.zip(values)] }
     end
+
+    # Checks that the imported hash has the required_keys, optionally also checks that the hash has
+    # no keys beyond those required when `allow_extra_keys` is false.
+    # returns `nil` if validation passes, or an error message if it fails
+    def validate_hash_import(hash, required_keys, allow_extra_keys) # :nodoc:
+      extra_keys = allow_extra_keys ? [] : hash.keys - required_keys
+      missing_keys = required_keys - hash.keys
+
+      return nil if extra_keys.empty? && missing_keys.empty?
+
+      if allow_extra_keys
+        <<-EOS
```

Comment:
Just a note: I didn't know what your official Ruby version support was, if 2.3+ then `~>>` could clean this up a bit.

Reply:
We haven’t officially dropped support for Rails 3.2, so  I think that means we support Ruby 2.2.